src/CoreBundle/private/library/classes/components/imageMagick/imageMagick.class.php

```diff
@@ -461,35 +432,38 @@ protected function GetNumberOfScenes()
     }
 
     /**
-     * resizes image to given proportions while keeping the aspect ratio.
-     *
-     * @param int $iWidth
-     * @param int $iHeight
-     *
-     * @return bool
+     * Resizes image to the specified width and height.
+     * Takes in account the image orientation set in exif or metadata and rotates the image if necessary.
      */
-    public function ResizeImage($iWidth, $iHeight)
+    public function ResizeImage(int $iWidth, int $iHeight): bool
     {
         $this->iThumbWidth = $iWidth;
         $this->iThumbHeight = $iHeight;
 
+        $aParameter = [];
+        
         $sImParamStrip = '';
         if ($this->bStrip) {
-            $sImParamStrip = ' -strip ';
+            $sImParamStrip = ' -strip';
+        }
+        
+        $orientation = $this->getImageOrientation();
+        
+        if (null !== $orientation) {
+            $aParameter[] = $this->getRotationCommandForOrientation((string) $orientation);
```

Comment:
der cast ist überflüssig, da getImageOrientation nur string oder null zurückgeben kann, und null bereits ausgeschlossen ist.

Reply:
You are right. Changed.